Sources/TootSDK/Models/ScheduledPostParams.swift

```diff
@@ -58,24 +58,41 @@ public struct ScheduledPostParams: Codable, Sendable {
     /// Unique post to prevent double posting
     public var idempotency: String?
     ///  ID of the post being replied to, if post is a reply.
-    public var inReplyToId: String?
+    public var inReplyToId: Int?
     /// (Pleroma) The MIME type of the post, it is transformed into HTML by the backend. You can get the list of the supported MIME types with the nodeinfo endpoint.
     public var contentType: String?
     /// (Pleroma) Will reply to a given conversation, addressing only the people who are part of the recipient set of that conversation. Sets the visibility to direct.
     public var inReplyToConversationId: String?
 
     enum CodingKeys: String, CodingKey {
         case text
-        case mediaIds = "media_ids"
+        case mediaIds
         case poll
-        case inReplyToId = "in_reply_to_id"
+        case inReplyToId
         case sensitive
-        case spoilerText = "spoiler_text"
+        case spoilerText
         case visibility
         case language
         case idempotency
-        case scheduledAt = "scheduled_at"
-        case contentType = "content_type"
-        case inReplyToConversationId = "in_reply_to_conversation_id"
+        case scheduledAt
+        case contentType
+        case inReplyToConversationId
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.text = try? container.decodeIfPresent(String.self, forKey: .text)
+        self.mediaIds = try? container.decodeIfPresent([String].self, forKey: .mediaIds)
+        self.poll = try? container.decodeIfPresent(CreatePoll.self, forKey: .poll)
+        self.inReplyToId = try? container.decodeIfPresent(Int.self, forKey: .inReplyToId)
+        // Mastodon incorrectly returns this as string
+        self.sensitive = try? container.decodeBoolFromString(forKey: .sensitive)
+        self.spoilerText = try? container.decodeIfPresent(String.self, forKey: .spoilerText)
+        self.visibility = (try? container.decode(Post.Visibility.self, forKey: .visibility)) ?? .public
```

Comment:
Unknown visibility should trigger a decoding error:

```suggestion
        self.visibility = try container.decode(Post.Visibility.self, forKey: .visibility)
```